OrbitClientData/ModuleData.cpp

```diff
@@ -8,14 +8,40 @@
 #include "OrbitClientData/FunctionUtils.h"
 #include "absl/synchronization/mutex.h"
 #include "capture_data.pb.h"
+#include "module.pb.h"
 
 using orbit_client_protos::FunctionInfo;
+using orbit_grpc_protos::ModuleInfo;
 
 bool ModuleData::is_loaded() const {
   absl::MutexLock lock(&mutex_);
   return is_loaded_;
 }
 
+void ModuleData::UpdateIfChanged(ModuleInfo info) {
+  absl::MutexLock lock(&mutex_);
+
+  CHECK(file_path() == info.file_path());
+
+  // TODO(171878807): Remove this as soon as a better way of distinguishing modules is implemented.
+  const bool build_id_different = !build_id().empty() && build_id() != info.build_id();
+  const bool no_build_id_but_module_different =
+      build_id().empty() &&
+      (name() != info.name() || file_size() != info.file_size() || load_bias() != info.load_bias());
+
+  if (build_id_different || no_build_id_but_module_different) {
+    LOG("Module %s changed.", file_path());
+    if (is_loaded_) {
+      LOG("Module %s contained symbols. Because the module changed, those are now removed.",
+          file_path());
+      functions_.clear();
+      is_loaded_ = false;
+    }
+  }
+
+  module_info_ = std::move(info);
```

Comment:
nit: maybe move this before the if and use early returns?

Reply:
Thank you!!! This was the hint I needed. ptal